intake/source/__init__.py

```diff
@@ -8,8 +8,60 @@
 import logging
 logger = logging.getLogger('intake')
 
+from ..utils import RegistryView
+
 # The registry is the mapping of plugin name (aka driver) to DataSource class
-registry = {}
+_registry = {}  # internal mutable registry
+registry = RegistryView(_registry)  # public, read-ony wrapper
+
+
+def register_driver(name, driver, overwrite=False):
+    """
+    Add a driver to intake.registry.
+
+    Parameters
+    ----------
+    name: string
+    driver: DataSource
+    overwrite: bool, optional
+        False by default.
+
+    Raises
+    ------
+    ValueError
+        If name collides with an existing name in the registry and overwrite is
+        False.
+    """
+    if name in _registry and not overwrite:
+        # If we are re-registering the same object, there is no problem.
+        original = _registry[name]
+        if original is driver:
+            return
+        raise ValueError(
+            f"The driver {driver} could not be registered for the "
+            f"name {name} because {_registry[name]} is already "
+            f"registered for that name. Use overwrite=True to force it.")
+    _registry[name] = driver
+
+
+def unregister_driver(name):
```

Comment:
I am not sure whether we actually need or want this one. Letting a package silently _remove_ a driver from underneath us isn't as likely to cause confusing bugs as silently _change_ a driver from underneath us, but it's still not great and I'm not sure there is a use case for this. Maybe `register_driver` with `overwrite` is enough.

Reply:
It is useful, at least for debugging. I don't expect too many end-users to need it, but I don't think it hurts.

Reply:
Ok, that works for me.